crates/analyzer/tests/snapshots/analysis__aug_assign.snap

```diff
@@ -264,7 +264,7 @@ note:
    ┌─ aug_assign.fe:65:5
    │  
 65 │ ╭     pub fn add_from_mem(a: u256, b: u256) -> u256 {
-66 │ │         let my_array: Array<u256, 10>
+66 │ │         let my_array: Array<u256, 10> = [0; 10]
```

Comment:
Adding this repeat expression feels a bit like a crutch to me. It doesn't seem overly useful for the user since it only allows a single value to be repeated. Did you consider to just initialize the array with default values on behalf of the user. Maybe that gets complicated if the array is over a complex type such as `Array<Customer, 10>`. However, in the future we could have a `Default` trait and if a user wants to initialize an array of `Array<Customer, 10>` then it will automatically use the values provided by `Customer::default()` or else error if `Customer` doesn't implement the `Default` trait. I'm just thinking out loud here...I'm not really against introducing the repeat expression but if in the future we find a way to get rid of it again or make it more useful I wouldn't be sad to see it go :smile:

Reply:
Ok, I just learned that one can use the same approach to initialize fixed size arrays in Rust. So, I guess it's fine. I still think that in the future it could be nice syntactical sugar if one could just write `let x: Array<Customer, 10>` and it would de-sugar into something like `let x: Array<Customer, 10> = [Customer::default(), 10]`. And if `T` does not implement the `Default` trait it would cause a compile error.